Build fails with GHC 9.10

```
Building library for calligraphy-0.1.6...
[ 1 of 19] Compiling Calligraphy.Prelude
[ 2 of 19] Compiling Calligraphy.Compat.GHC
[ 3 of 19] Compiling Calligraphy.Util.Lens
[ 4 of 19] Compiling Calligraphy.Compat.Lib
[ 5 of 19] Compiling Calligraphy.Phases.Search
[ 6 of 19] Compiling Calligraphy.Util.LexTree
[ 7 of 19] Compiling Calligraphy.Util.Optparse
[ 8 of 19] Compiling Calligraphy.Util.Printer
[ 9 of 19] Compiling Calligraphy.Compat.Debug
[10 of 19] Compiling Calligraphy.Util.Types
[11 of 19] Compiling Calligraphy.Phases.Render.Common
[12 of 19] Compiling Calligraphy.Phases.Render.Mermaid
[13 of 19] Compiling Calligraphy.Phases.Render.GraphViz
[14 of 19] Compiling Calligraphy.Phases.Parse
src/Calligraphy/Phases/Parse.hs:105:23: error: [GHC-83865]
    • Couldn't match type ‘ghc-internal-9.1001.0:GHC.Internal.Word.Word64’
                     with ‘Int’
      Expected: GHC.Types.Unique.Unique -> Int
        Actual: GHC.Types.Unique.Unique
                -> ghc-internal-9.1001.0:GHC.Internal.Word.Word64
    • In the first argument of ‘(.)’, namely ‘GHC.getKey’
      In the second argument of ‘(.)’, namely
        ‘GHC.getKey . GHC.nameUnique’
      In the expression: GHCKey . GHC.getKey . GHC.nameUnique
    |
105 | ghcNameKey = GHCKey . GHC.getKey . GHC.nameUnique
    |                       ^^^^^^^^^^

[15 of 19] Compiling Calligraphy.Phases.NodeFilter
[16 of 19] Compiling Calligraphy.Phases.EdgeCleanup
[17 of 19] Compiling Calligraphy.Phases.DependencyFilter
[18 of 19] Compiling Paths_calligraphy
Error: [Cabal-7125]
Failed to build calligraphy-0.1.6 (which is required by exe:calligraphy from calligraphy-0.1.6). See the build log above for details.
```

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -40,7 +40,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        resolver: [ nightly, lts-20, lts-19, lts-18, lts-17, lts-16, lts-15 ]
+        resolver: [ nightly, lts-23, lts-22, lts-21, lts-20, lts-19, lts-18, lts-17, lts-16, lts-15 ]
         # macos-13 is a workaround for https://github.com/haskell-actions/setup/issues/77
         os: [ macos-13, ubuntu-latest ]
     runs-on: ${{ matrix.os }}

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -40,7 +40,8 @@ jobs:
       fail-fast: false
       matrix:
         resolver: [ nightly, lts-20, lts-19, lts-18, lts-17, lts-16, lts-15 ]
-        os: [ macos-latest, ubuntu-latest ]
+        # macos-13 is a workaround for https://github.com/haskell-actions/setup/issues/77
+        os: [ macos-13, ubuntu-latest ]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
@@ -58,7 +59,7 @@ jobs:
       fail-fast: false
       matrix:
         ghc: [ '8.8.4', '8.10.7', '9.0.2', '9.2.8', '9.4.8', '9.6.6', '9.8.4', '9.10.1' ]
-        os: [ ubuntu-latest, macos-latest ]
+        os: [ ubuntu-latest, macos-13 ]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -22,7 +22,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: [ ghc910, ghc98, ghc96, ghc94, ghc92, ghc90, ghc810, ghc88 ]
+        ghc: [ ghc910, ghc98, ghc96 ]
+        # GHC 9.4 and earlier have issues with the `unix` package that I don't feel like figuring out
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -12,7 +12,7 @@ jobs:
     name: Nix default
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: cachix/install-nix-action@v19
       - run: nix build -L
       - run: nix develop -L -c echo "All good"
@@ -22,10 +22,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: [ ghc8107, ghc902, ghc927, ghc945, ghc962 ]
+        ghc: [ ghc910, ghc98, ghc96, ghc94, ghc92, ghc90, ghc810, ghc88 ]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: cachix/install-nix-action@v19
       - name: nix build
         run: nix build -L .#calligraphyFor.${{ matrix.ghc }}
@@ -43,8 +43,8 @@ jobs:
         os: [ macos-latest, ubuntu-latest ]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: haskell/actions/setup@v2
+      - uses: actions/checkout@v4
+      - uses: haskell-actions/setup@v2
         with:
           enable-stack: true
       - run: stack init --resolver ${{ matrix.resolver }}
@@ -57,12 +57,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: [ '8.8.4', '8.10.7', '9.0.2', '9.2.7', '9.4.5', '9.6.2' ]
+        ghc: [ '8.8.4', '8.10.7', '9.0.2', '9.2.8', '9.4.8', '9.6.6', '9.8.4', '9.10.1' ]
         os: [ ubuntu-latest, macos-latest ]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: haskell/actions/setup@v2
+      - uses: actions/checkout@v4
+      - uses: haskell-actions/setup@v2
         with:
           ghc-version: ${{ matrix.ghc }}
       - uses: actions/cache@v3

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.1.7
+### [Added]
+- [#46] GHC 9.8, 9.10 support
+
 ## 0.1.6
 ### [Changed]
 - Internal change; partially undid the Prelude structure implemented in [#22] and [#27], since it was causing issues (see [#32])

--- a/calligraphy.cabal
+++ b/calligraphy.cabal
@@ -1,6 +1,6 @@
 cabal-version:   2.4
 name:            calligraphy
-version:         0.1.6
+version:         0.1.7
 license:         BSD-3-Clause
 build-type:      Simple
 license-file:    LICENSE
@@ -8,7 +8,14 @@ author:          Jonas Carpay
 maintainer:      Jonas Carpay <jonascarpay@gmail.com>
 copyright:       2022 Jonas Carpay
 tested-with:
-  GHC ==8.8.4 || ==8.10.7 || ==9.0.2 || ==9.2.7 || ==9.4.5 || ==9.6.2
+  GHC ==8.8.4
+   || ==8.10.7
+   || ==9.0.2
+   || ==9.2.8
+   || ==9.4.8
+   || ==9.6.6
+   || ==9.8.4
+   || ==9.10.1
 
 extra-doc-files:
   CHANGELOG.md

--- a/flake.nix
+++ b/flake.nix
@@ -1,7 +1,7 @@
 {
   description = "calligraphy";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-23.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-24.11";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
   outputs = inputs:

--- a/flake.nix
+++ b/flake.nix
@@ -22,12 +22,14 @@
         let
 
           per-compiler = f: pkgs.lib.genAttrs [
-            "ghc962"
-            "ghc945"
-            "ghc927"
-            "ghc902"
-            "ghc8107"
-            "ghc884"
+            "ghc910"
+            "ghc98"
+            "ghc96"
+            "ghc94"
+            "ghc92"
+            "ghc90"
+            "ghc810"
+            "ghc88"
           ]
             (ghc: f pkgs.haskell.packages.${ghc});
 

--- a/src/Calligraphy/Phases/Parse.hs
+++ b/src/Calligraphy/Phases/Parse.hs
@@ -102,7 +102,7 @@ ppLexTree = foldLexTree (pure ()) $ \ls l decl m r rs -> do
   rs
 
 ghcNameKey :: GHC.Name -> GHCKey
-ghcNameKey = GHCKey . GHC.getKey . GHC.nameUnique
+ghcNameKey = GHCKey . fromIntegral . GHC.getKey . GHC.nameUnique
 
 newtype ParsePhaseDebugInfo = ParsePhaseDebugInfo {modulesLexTrees :: [(String, LexTree Loc RawDecl)]}
 

